### ISSUE TITLE
Add support for actor events (similar to service fabric)

### ISSUE THREAD

**Issue by darraghjones** [dapr/dapr#926]
## In what area(s)?
/area runtime

## Describe the feature
I'd like to see a feature similar to service fabric actor events, i.e. 
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-actors-events

In my use-case, an actor in my system represents a game of Connect4 between two players. When one player moves, I'd like to notify the other player.

**Comment by drk9999** [comment]
#814 would allow for this to be implemented at the SDK Level. Please comment on that issue (#814) as well. 

The reason IMO it seams like a better idea to implement this at the SDK level, is because each language has it's own pattern for events, if any. 

In the case of the .NET SDK, it could be implemented by the proxy base class that uses reflection, to call a remote method to register a unique topic name in pub/sub if the <IActorX> proxy had any type of delegate members. By the same token, the Actor base class could implement the method to register this unique topic name, and keep a dictionary of all of the topic names for any clients by instance that needed to get notified when the event an event is invoked in the actor class. 


In short, I fully agree this is a requirement, since I can't really see more then about a 10-20% use case for actors that didn't need some way of raising an event back to the client

**Comment by darraghjones** [comment]
I don't feel like it should be implemented at the SDK level, as it would not be a language agnostic feature. 

However, it's not clear to me _how_ it should be implemented. I wonder if server sent events would provide the necessary underlying functionality while sticking close to the design philosophy of Dapr.

**Comment by darraghjones** [comment]
Actually, gRPC streams might be a good choice for this...
